### PR TITLE
Order quantity promotion rule (2-4-stable)

### PR DESCRIPTION
**Pull request by AlessioRocco**
This is the same of https://github.com/spree/spree/pull/5957 but for 2-4-stable, the only thing that changed is the rule view.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://alessiorocco-2-4-stable-order-quantity-promotion.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://alessiorocco-2-4-stable-order-quantity-promotion.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: core/app/models/spree/promotion/rules/item_quantity.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  class Promotion
+    module Rules
+      class ItemQuantity < PromotionRule
+        preference :quantity_min, :integer, default: 1
+        preference :operator_min, :string, default: '>'
+        preference :quantity_max, :integer, default: 1000
+        preference :operator_max, :string, default: '<'
+
+        OPERATORS_MIN = ['gt', 'gte']
+        OPERATORS_MAX = ['lt', 'lte']
```
Comment (jhawthorn) [inline core/app/models/spree/promotion/rules/item_quantity.rb]: Is this necessary? You can't have `.5` of a line item and this adds a fair bit of logic.

**Comment by athal7** [comment]
this seems like a new feature to me. IMHO this should not be introduced in a stable branch.

**Comment by gmacdougall** [comment]
I agree with @athal7 that this feature should not be targeted towards Spree 2-4-stable. If you need this feature in Spree 2.4, I recommend using it in your local fork of Spree for the time being.

**Comment by JDutil** [comment]
Closing in favor of just #5957

**Pull request closed without merging**